src/gui/editors/SongEditor.cpp

```diff
@@ -756,22 +756,29 @@ void SongEditor::updatePosition( const TimePos & t )
 	const auto widgetWidth = compactTrackButtons ? DEFAULT_SETTINGS_WIDGET_WIDTH_COMPACT : DEFAULT_SETTINGS_WIDGET_WIDTH;
 	const auto trackOpWidth = compactTrackButtons ? TRACK_OP_WIDTH_COMPACT : TRACK_OP_WIDTH;
 
-	if( ( m_song->isPlaying() && m_song->m_playMode == Song::PlayMode::Song
-		  && m_timeLine->autoScroll() == TimeLineWidget::AutoScrollState::Enabled) ||
-							m_scrollBack == true )
+	if ((m_song->isPlaying() && m_song->m_playMode == Song::PlayMode::Song)
+							|| m_scrollBack == true)
 	{
 		m_smoothScroll = ConfigManager::inst()->value( "ui", "smoothscroll" ).toInt();
 		const int w = width() - widgetWidth
 							- trackOpWidth
 							- contentWidget()->verticalScrollBar()->width(); // width of right scrollbar
-		if( t > m_currentPosition + w * TimePos::ticksPerBar() /
-							pixelsPerBar() )
+		
+		if (m_timeLine->autoScroll() == TimeLineWidget::AutoScrollState::Stepped)
 		{
-			animateScroll( m_leftRightScroll, t.getBar(), m_smoothScroll );
+			if (t > m_currentPosition + w * TimePos::ticksPerBar()
+								/ pixelsPerBar())
+			{
+				animateScroll(m_leftRightScroll, t.getTicks(), m_smoothScroll);
+			}
+			else if (t < m_currentPosition)
+			{
+				animateScroll(m_leftRightScroll, t.getTicks(), m_smoothScroll);
+			}
```

Comment:
I think this is simpler and should work the same way. Feel free to change the name of `nextPosition` if that's not accurate.
```suggestion
			const auto nextPosition = m_currentPosition + w * TimePos::ticksPerBar() / pixelsPerBar();
			if (t <= nextPosition && t >= m_currentPosition) { return; }
			animateScroll(m_leftRightScroll, t.getTicks(), m_smoothScroll);
```